File: hw1/Codes/Oztunc_Zeynep.dart

void main() {
 //initializing the array
var newArr = {
'a': 10,
'b': 20,
'c': 30,
'd': 40,
'e': 50,
'f': 60,  
};
print(newArr);


//get the value for the given key
print("The value for the key a is:");
print(newArr['a']);

// add a new element
newArr['g '] =70;
print(newArr);
  
//remove an element of the array
newArr.remove('f');
print(newArr);

//modify the value of an existing element
print("The initial value of e is : ");
print(newArr['e']);
newArr['e'] =100;
print("The value of e after modification is: ");
print(newArr['e']);

//search for the existence of a key
print(newArr.containsKey("a"));
print(newArr.containsKey("z"));

//search for the existence of a value
int value=20;
print("Does the value $value  exist in the array?");
print(newArr.containsValue(20));

//loop through the array
  for (String key in newArr.keys){
    foo(key,newArr);
 }

//foo function
}
void foo(key,newArr){
   print("The value for the key $key is: ");
   print(newArr[key]);
}

File: hw1/Codes/Oztunc_Zeynep.rs
//rustc 1.39.0

fn main() {

//initializing the array
    use std::collections::HashMap;
    //let mut newArr = HashMap::new();
    let mut newArr: HashMap<&str, i32> =
    [("a", 4),
    ("b", 3),
    ("c", 4),
    ("d", 2)].iter().cloned().collect();
    println!("{:?}", newArr);

//get the value for the given key
    println!("{}",newArr["b"]);

// add a new element
    println!("Adding a new element with key e");
    newArr.insert("e", 50);
    println!("{:?}", newArr);

//remove an element of an array
     println!("Remove the element with key a");
     newArr.remove("a");
     println!("{:?}", newArr);

//modify the value of an existing element
    println!("Modify the element wih key e");
    newArr.insert("e", 100);
    println!("{:?}", newArr);

//search for the existence of a key
println!("Does the key b exist?");
println!("{}", newArr.contains_key("b"));

//search for the existence of a value
println!("Does the value 40 exist?");
let value_exists = newArr.values().any(|&x| x > 0);
    if value_exists{
            println!("The value exists");
    }
    else{
        println!("The value does not exist");
    }

//loop through the array
 for (key, value) in &newArr {
       foo(key.to_string(),*value);
  }
}
//foo function
fn foo(key:String,value:i32){
     print!("{}: {} ", key, value);
}
